query-engine/driver-adapters/js/smoke-test-js/src/libquery/libquery.ts

```diff
@@ -1,170 +1,423 @@
+              "count": true
+            }
+          }
+        }
+      )
+    })
 
-    await this.doQuery(
-      {
-        "action": "deleteMany",
-        "modelName": "type_test_2",
-        "query": {
-          "arguments": {
-            "where": {}
-          },
-          "selection": {
-            "count": true
+    it('create/delete parent and child', async () => {
+      /* Delete all child and parent records */
+
+      // Queries: [
+      //   'SELECT `cf-users`.`Child`.`id` FROM `cf-users`.`Child` WHERE 1=1',
+      //   'SELECT `cf-users`.`Child`.`id` FROM `cf-users`.`Child` WHERE 1=1',
+      //   'DELETE FROM `cf-users`.`Child` WHERE (`cf-users`.`Child`.`id` IN (?) AND 1=1)'
+      // ]
+      await doQuery(
+        {
+          "modelName": "Child",
+          "action": "deleteMany",
+          "query": {
+            "arguments": {
+              "where": {}
+            },
+            "selection": {
+              "count": true
+            }
+          }
+        }
+      )
+
+      // Queries: [
+      //   'SELECT `cf-users`.`Parent`.`id` FROM `cf-users`.`Parent` WHERE 1=1',
+      //   'SELECT `cf-users`.`Parent`.`id` FROM `cf-users`.`Parent` WHERE 1=1',
+      //   'DELETE FROM `cf-users`.`Parent` WHERE (`cf-users`.`Parent`.`id` IN (?) AND 1=1)'
+      // ]
+      await doQuery(
+        {
+          "modelName": "Parent",
+          "action": "deleteMany",
+          "query": {
+            "arguments": {
+              "where": {}
+            },
+            "selection": {
+              "count": true
+            }
+          }
+        }
+      )
+
+      /* Create a parent with some new children, within a transaction */
+
+      // Queries: [
+      //   'INSERT INTO `cf-users`.`Parent` (`p`,`p_1`,`p_2`,`id`) VALUES (?,?,?,?)',
+      //   'INSERT INTO `cf-users`.`Child` (`c`,`c_1`,`c_2`,`parentId`,`id`) VALUES (?,?,?,?,?)',
+      //   'SELECT `cf-users`.`Parent`.`id`, `cf-users`.`Parent`.`p` FROM `cf-users`.`Parent` WHERE `cf-users`.`Parent`.`id` = ? LIMIT ? OFFSET ?',
+      //   'SELECT `cf-users`.`Child`.`id`, `cf-users`.`Child`.`c`, `cf-users`.`Child`.`parentId` FROM `cf-users`.`Child` WHERE `cf-users`.`Child`.`parentId` IN (?)'
+      // ]
+      await doQuery(
+        {
+          "modelName": "Parent",
+          "action": "createOne",
+          "query": {
+            "arguments": {
+              "data": {
+                "p": "p1",
+                "p_1": "1",
+                "p_2": "2",
+                "childOpt": {
+                  "create": {
+                    "c": "c1",
+                    "c_1": "foo",
+                    "c_2": "bar"
+                  }
+                }
+              }
+            },
+            "selection": {
+              "p": true,
+              "childOpt": {
+                "selection": {
+                  "c": true
+                }
+              }
+            }
           }
         }
+      )
+
+      /* Delete the parent */
```

Comment:
Should we do a find query before and after delete? Currently it's only testing it's not throwing but not that the queries work as expected.

Reply:
For this kind of smoke test, I'd find that overkill. These tests are meant to be quick checks to ensure we didn't create any significant regression when manipulating strictly driver-adapters-related code. The shape of the result set per se is checked by the rest of the Rust/TypeScript tests.